packages/core/src/types/runtime.ts

```diff
@@ -64,6 +64,15 @@ export interface DataAccessor<T = DefaultType> {
   nextBatch: (batchSize: number) => Promise<Array<Sample<T>> | null>;
   seek: (pos: number) => Promise<void>;
 }
+export interface RandomDataAccessor<T = DefaultType> extends DataAccessor {
+  randomMap: Array<number>;
+  dataLength: number;
+  init: (SourceMeta: ImageDataSourceMeta) => void;
```

Comment:
They are not a part of the api, but the implement, I think it's better to move them into class.